### PR TITLE
Redeem bounty

### PR DESCRIPTION
**Pull request by dev-matan-tsuberi**

### CODE REVIEW

File: src/actions/arcActions.ts [hunk 1]
```diff
@@ -324,7 +325,7 @@ async function getProposalDetails(dao: Arc.DAO, votingMachineInstance: Arc.Genes
     votesYes: Util.fromWei(yesVotes).toNumber(),
     votesNo: Util.fromWei(noVotes).toNumber(),
     winningVote: Number(proposalDetails[9]),
-    threshold: Util.fromWei(await votingMachineInstance.getThreshold({avatar: dao.avatar.address, proposalId})).toNumber()
+    threshold: Util.fromWei(new BigNumber(await votingMachineInstance.getThreshold({avatar: dao.avatar.address, proposalId}))).toNumber()
```
Comment (dkent600) [inline src/actions/arcActions.ts]: `getThreshold` should already be returning a BigNumber

File: src/actions/arcActions.ts [hunk 2]
```diff
@@ -381,6 +382,15 @@ async function getRedemptions(avatarAddress: string, votingMachineInstance: Arc.
 
   const proposalId = proposal.proposalId;
 
+  const stakingTokenAddress = await votingMachineInstance.contract.stakingToken();
+  const StandardToken = await Arc.Utils.requireContract('StandardToken');
+  const stakingToken = await StandardToken.at(stakingTokenAddress);
+  const avatarStakingTokenBalance = Util.fromWei(await stakingToken.balanceOf(avatarAddress)).toNumber();
```
Comment (dkent600) [inline src/actions/arcActions.ts]: Can you do these four lines in a single line using `votingMachineInstance.getTokenBalances`?

File: src/actions/arcActions.ts [hunk 3]
```diff
@@ -607,7 +628,7 @@ export function createProposal(daoAvatarAddress: string, title: string, descript
         votesYes: 0,
         votesNo: 0,
         winningVote: 0,
-        threshold: Util.fromWei(await votingMachineInstance.getThreshold({avatar: daoAvatarAddress, proposalId})).toNumber()
+        threshold: Util.fromWei(new BigNumber(await votingMachineInstance.getThreshold({avatar: daoAvatarAddress, proposalId}))).toNumber()
```
Comment (dkent600) [inline src/actions/arcActions.ts]: `getThreshold` should already be returning a BigNumber

Comment (dev-matan-tsuberi) [reply src/actions/arcActions.ts]: fixed: https://github.com/daostack/alchemy/pull/81/commits/ee837c9797978fc6976021364697bca1ff0c86d0

File: src/actions/arcActions.ts (hunk 2 above)
Comment (dev-matan-tsuberi) [reply src/actions/arcActions.ts]: fixed: https://github.com/daostack/alchemy/pull/81/commits/d1483428ad4834d4151474d0afc525e89d29d400

File: src/actions/arcActions.ts (hunk 1 above)
Comment (dev-matan-tsuberi) [reply src/actions/arcActions.ts]: fixed: https://github.com/daostack/alchemy/pull/81/commits/ee837c9797978fc6976021364697bca1ff0c86d0

**Comment by dev-matan-tsuberi** [comment]
[Add redeeming of staking bounty](https://trello.com/c/KGaWCKdL/206-add-redeeming-of-staking-bounty)

**Pull request merged**